File: compatLib/compatLibVU/src/main/java/app/lawnchair/compatlib/fourteen/QuickstepCompatFactoryVU.java
package app.lawnchair.compatlib.fourteen;

import android.window.RemoteTransition;
import androidx.annotation.NonNull;
import androidx.annotation.RequiresApi;
import app.lawnchair.compatlib.ActivityManagerCompat;
import app.lawnchair.compatlib.ActivityOptionsCompat;
import app.lawnchair.compatlib.RemoteTransitionCompat;
import app.lawnchair.compatlib.thirteen.QuickstepCompatFactoryVT;

@RequiresApi(34)
public class QuickstepCompatFactoryVU extends QuickstepCompatFactoryVT {

    @NonNull
    @Override
    public ActivityManagerCompat getActivityManagerCompat() {
        return new ActivityManagerCompatVU();
    }

    @NonNull
    @Override
    public ActivityOptionsCompat getActivityOptionsCompat() {
        return new ActivityOptionsCompatVU();
    }

    @NonNull
    @Override
    public RemoteTransitionCompat getRemoteTransitionCompat() {
        return RemoteTransition::new;
    }
}

File: compatLib/compatLibVU/src/app/lawnchair/compatlib/fourteen/QuickstepCompatFactoryVU.kt
package app.lawnchair.compatlib.fourteen;

import android.app.IApplicationThread
import android.window.IRemoteTransition
import android.window.RemoteTransition
import app.lawnchair.compatlib.ActivityManagerCompat
import app.lawnchair.compatlib.ActivityOptionsCompat
import app.lawnchair.compatlib.RemoteTransitionCompat
import app.lawnchair.compatlib.thirteen.QuickstepCompatFactoryVT

class QuickstepCompatFactoryVU : QuickstepCompatFactoryVT() {

    override fun getActivityManagerCompat(): ActivityManagerCompat {
        return ActivityManagerCompatVU()
    }

    override fun getActivityOptionsCompat(): ActivityOptionsCompat {
        return ActivityOptionsCompatVU()
    }

    override fun getRemoteTransitionCompat(): RemoteTransitionCompat {
        return object : RemoteTransitionCompat() {
            override fun getRemoteTransition(
                remoteTransition: IRemoteTransition,
                appThread: IApplicationThread?,
                debugName: String?
            ): RemoteTransition {
                return RemoteTransition(remoteTransition, appThread, debugName)
            }
        }
    }
}
